transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannel.java

```diff
@@ -463,6 +428,26 @@ protected Object filterOutboundMessage(Object msg) {
                 "unsupported message type: " + StringUtil.simpleClassName(msg) + EXPECTED_TYPES);
     }
 
+    /**
+     * Checks if the specified buffer has memory address or is composed of n(no more than IOV_MAX) NIO direct buffers.
+     * (We check this because otherwise we need to make it a new direct buffer.)
+     */
+    private static boolean isBufferNoNeedToConvert(ByteBuf byteBuf) {
+        if (byteBuf.hasMemoryAddress()) {
+            return true;
+        }
+
+        // less then 1024 buffers for gathering writes
+        if (byteBuf.isDirect() && byteBuf.nioBufferCount() <= IOV_MAX) {
+            return true;
+        } else {
+            if (byteBuf.isDirect()) {
+                return true;
```

Comment:
this looks like a bug. Method will return `true` regardless of `nioBufferCount()`.